platform/nucleo-board-stm32f334r8/platform.h

```diff
@@ -0,0 +1,75 @@
+/**
+ * Platform support for the STM32-Nucleo64 Board F334R8
+ * http://www.st.com/en/evaluation-tools/nucleo-f334r8.html
+ */
+
+#ifndef PLATFORM_H
+#define PLATFORM_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+#include <stddef.h>
+
+#define PLATFORM_DEVICE_CLASS "nucleo-board-stm32f334r8"
+#define FLASH_PAGE_SIZE 0x0800 // 2K
+#define CONFIG_PAGE_SIZE FLASH_PAGE_SIZE
+
+// Onboard LED
+#define GPIO_PORT_LED2  GPIOA
+#define GPIO_PIN_LED2   GPIO5
+
+// CAN pins
+#define GPIO_PORT_CAN_RX    GPIOB
+#define GPIO_PIN_CAN_RX     GPIO8
+#define GPIO_PORT_CAN_TX    GPIOB
+#define GPIO_PIN_CAN_TX     GPIO9
+#define GPIO_AF_CAN         GPIO_AF9
+
+/*
+ * Many CAN transceivers have an enable input,
+ * which needs to be driven HIGH in order
+ * for the transceiver to become operational.
+ * Define USE_CAN_ENABLE to set the pin
+ * defined below to HIGH upon startup.
+ */
+//#define USE_CAN_ENABLE
+#ifdef USE_CAN_ENABLE
+#define GPIO_PORT_CAN_ENABLE    GPIOA
+#define GPIO_PIN_CAN_ENABLE     GPIO8
+#endif
+
+// Import symbols from linker script
+extern uint8_t config_page_buffer[CONFIG_PAGE_SIZE];
+extern int application_address, application_size, config_page1, config_page2;
+
+
+// TODO: Superfluous(?) inline functions
```

Comment:
Yes, it could be done that way. Probably some legacy code, I can't remember any particular reason it was done that way. It was introduced in 7a2fb0f5 by @nuft maybe he remembers why he did it that way ?